### DISCUSSION TITLE
“View File” in a pull request from a fork fails "Unable to view file"

### DISCUSSION THREAD

**Discussion by KalleOlaviNiemitalo** [community/community#132347] [Mobile]
I open <https://github.com/jenkinsci/azure-ad-plugin/pull/596> in the GitHub app and touch “12 files changed”.  The app shows the README.md changes first.  I then touch the ellipsis button at the right side of the file name and choose “View File”:

![image](https://github.com/user-attachments/assets/4486c1bb-db46-46b6-aef7-43a88f96f134)

However, the app then shows “Unable to view file”:

![image](https://github.com/user-attachments/assets/081e7c1b-c21d-42bd-8bcf-d364f506d461)

I touch “Open in browser”, which opens <https://github.com/jenkinsci/azure-ad-plugin/blob/add-client-certificate-support/README.md> in Safari, but that page also shows an error “Cannot find a valid ref”:

![image](https://github.com/user-attachments/assets/5f8b730d-4de3-4964-86f9-ca70c1cd9a34)

Indeed it’s the wrong URL: the source branch "add-client-certificate-support" of the pull request is not in the "jenkinsci/azure-ad-plugin" repository; it is instead in the "itzikbekelmicrosoft/azure-ad-plugin" fork.  The correct URL would thus be <https://github.com/jenkinsci/azure-ad-plugin/blob/488f3fe7ea2d3fb3d83a72e61f5398d48462f59f/README.md> (what the web site links to) or <https://github.com/itzikbekelmicrosoft/azure-ad-plugin/blob/add-client-certificate-support/README.md> (just replacing the owner and repository segments in the URL path).

<details>
<summary>ℹ️ Additional Debug Info</summary>
GitHub for iOS, Version 1.168.0 (135483217)
Device: iPhone SE (2nd generation) (iOS 17.5.1)
</details>

**Comment by HenSquared** [comment]
@KalleOlaviNiemitalo, 

Thank you for this discussion. Your post is really in-depth and made it really easy for me to reproduce the issue on my device. I've filed an issue internally to follow up and fix this. We will update here if we need extra info. 

Thank you. :)
